packages/google-pay-integration/src/gateways/google-pay-gateway.ts

```diff
@@ -0,0 +1,210 @@
+import { round } from 'lodash';
+
+import {
+    AddressRequestBody,
+    BillingAddressRequestBody,
+    guard,
+    InvalidArgumentError,
+    NotInitializedError,
+    NotInitializedErrorType,
+    PaymentIntegrationService,
+    PaymentMethod,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import isGooglePayCardNetworkKey from '../guards/is-google-pay-card-network-key';
+import {
+    GooglePayCardDataResponse,
+    GooglePayCardNetwork,
+    GooglePayCardParameters,
+    GooglePayFullBillingAddress,
+    GooglePayGatewayParameters,
+    GooglePayInitializationData,
+    GooglePayMerchantInfo,
+    GooglePaySetExternalCheckoutData,
+    GooglePayTransactionInfo,
+} from '../types';
+
+export default class GooglePayGateway {
+    private _getPaymentMethodFn?: () => PaymentMethod<GooglePayInitializationData>;
+
+    constructor(
+        private _gatewayIdentifier: string,
+        private _paymentIntegrationService: PaymentIntegrationService,
+    ) {}
+
+    mapToShippingAddressRequestBody({
+        shippingAddress,
+    }: GooglePayCardDataResponse): AddressRequestBody | undefined {
+        if (!shippingAddress) {
+            return;
+        }
+
+        const { company = '', phone = '' } =
+            this._paymentIntegrationService.getState().getShippingAddress() || {};
+
+        return this._mapToAddressRequestBody(shippingAddress, company, phone);
+    }
+
+    mapToBillingAddressRequestBody(
+        response: GooglePayCardDataResponse,
+    ): BillingAddressRequestBody | undefined {
+        const { billingAddress } = response.paymentMethodData.info;
+
+        if (!billingAddress) {
+            return;
+        }
+
+        const {
```

Comment:
Can we move this to the `_mapToAddressRequestBody` method itself? Rather than passing these values are parameters?

Reply:
Since `_mapToAddressRequestBody` is used to map both shipping and billing addresses, I didn't find it convenient to move it. The function does not know the source of the data, its only function is to map.